src/middlewared/middlewared/plugins/network.py

```diff
@@ -209,6 +226,9 @@ async def do_update(self, data):
         new_config = config.copy()
         is_ha = True
         sync_group_mappings = False
+        srv = config['service_announcement'] | data.get('service_announcement', {})
```

Comment:
How would i got about removing a service announcement which i had enabled before ?

Reply:
```
>>> d1 = {"netbios": True, "mdns": False}
>>> d2 = {"netbios": False, "mdns": False}
>>> d1 | d2
{'netbios': False, 'mdns': False}
```
Maybe not understanding question

Reply:
@anodos325 how does this behave when `service_announcement` is declared at L185 like
```
            Dict(
                'service_announcement',
                Bool('netbios', default=False),
                Bool('mdns', default=True),
                Bool('wsd', default=True),
            ),
```
?

Reply:
Well, goal there was to make sure that dict was normalized before being converted to JSON and written to DB. I'll switch normalization to inside update() method so that we go <defaults> | <what's on disk> | <payload>.

Reply:
Thank you for clarifying @anodos325 :P I was looking at it the wrong way :D